### PR TITLE
Update ElasticRetriever for cursor pagination

### PR DESCRIPTION
**Pull request by JarrodBaker**
The implementation is a little more complex than was strictly necessary, but I wanted to avoid breaking the contract with the front-end who have been passing an array of strings to facilitate pagination. 

Previously the front-end needed to provide an array of Strings. Elasticsearch changed their interface and now require an Int and a String. The front-end doesn't not need to make any changes, as this implementation stashes the `(Int, String)` tuple and returns a key to the front-end which can be returned to redeem the pagination tokens which are then passed on to Elasticsearch. 

As discussed in the backend meeting today, this fixes the problem for now. The other obvious option is having the front-end be responsible for managing the cursor tokens.

### CODE REVIEW

File: app/models/SearchAfterCache.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import play.api.inject.SimpleModule
+import play.api.{Logging, inject}
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+
+class CacheModule extends SimpleModule(inject.bind[SearchAfterCache].toSelf.eagerly())
+
+object SearchAfterCache extends Logging {
```
Comment (mkarmona) [inline app/models/SearchAfterCache.scala]: @JarrodBaker be careful with this. we should not implementing this ourselves. Woud you mind check this already way to do it as Play already provide this cache mechanism for you (Caffeine) https://www.playframework.com/documentation/2.8.x/ScalaCache

Comment (mkarmona) [reply app/models/SearchAfterCache.scala]: Using internal simple cache for caffeine would be enough

Comment (mkarmona) [reply app/models/SearchAfterCache.scala]: Ehcache will be obsoleted and caffeine is the new one

**Review by mkarmona — CHANGES_REQUESTED** [review]

File: app/models/SearchAfterCache.scala (hunk 1 above)
Comment (JarrodBaker) [reply app/models/SearchAfterCache.scala]: Thanks for the tip: I've deleted by home grown stuff and have implemented using Caffeine.

**Review by mkarmona — APPROVED** [review]

**Pull request merged**